ui/src/containers/SolutionList.js

```diff
@@ -42,56 +105,289 @@ const SolutionsList = props => {
-      dataKey: 'versions',
-      renderer: versions =>
-        versions.map((version, index) => {
-          return version.deployed && version.ui_url ? (
-            <a href={version.ui_url} key={`deployed_version_${index}`}>
-              {version.version}
-            </a>
-          ) : null;
-        }),
-      flexGrow: 1
-    }
+      label: intl.messages.name,
+      dataKey: 'name',
+    },
+    {
+      label: intl.messages.description,
+      dataKey: 'description',
+      flexGrow: 1,
+    },
+    {
+      label: intl.messages.solutions,
+      dataKey: 'solutions',
+      renderer: (solutions, row) => {
+        return (
+          <div>
+            <span>
+              {solutions.map((solution, idx) => {
+                const solutionRow = sortedSolutions.find(
+                  s => s.name === solution.name,
+                );
+
+                const solutionVersion = solutionRow?.versions?.find(
+                  v => v.version === solution.version,
+                );
+
+                return (
+                  <ButtonContainer
+                    key={`solution_${idx}`}
+                    marginLeft={idx !== 0}
+                  >
+                    <Button
+                      size="smaller"
+                      text={`${solution.name} ${solution.version}`}
+                      icon={<i className="fas fa-external-link-alt" />}
+                      onClick={() => {
+                        if (solutionRow && solutionVersion) {
+                          const url = `${solutionVersion.ui_url}/environments/${row.name}/version/${solution.version}/prepare`;
+                          window.open(url, '_blank');
+                        }
+                      }}
+                    ></Button>
+                  </ButtonContainer>
+                );
+              })}
+            </span>
+            <ButtonContainer marginLeft={solutions.length !== 0}>
+              <Button
+                size="smaller"
+                icon={<i className="fas fa-plus" />}
+                onClick={() => {
+                  setSelectedEnvironment(row.name);
+                  setisAddSolutionModalOpen(true);
+                }}
+              />
+            </ButtonContainer>
+          </div>
+        );
+      },
+      flexGrow: 1,
+    },
   ];
 
   const onSort = ({ sortBy, sortDirection }) => {
     setSortBy(sortBy);
     setSortDirection(sortDirection);
   };
 
-  const solutionsSortedList = sortSelector(solutions, sortBy, sortDirection);
+  const sortedSolutions = sortSelector(solutions, sortBy, sortDirection) ?? [];
+
+  const formattedEnvironments = environments.map(environment => {
+    return {
+      name: environment?.metadata?.name ?? '',
+      description: environment?.spec?.description ?? '',
+      solutions: environment?.spec?.solutions ?? [],
+    };
+  });
+
+  const firstSolution = sortedSolutions[0] && sortedSolutions[0].name;
+
+  const firstVersion =
```

Comment:
Can't use the same syntax for such accesses? Something like `sortedSolution?[0]?.versions?[0]?.version` would be nice

Reply:
Unfortunately, we can't. It's an access of an array element not an object field.
Optional chaining is for this kind of case : 
```javascript
let a = { 'dynamic_field': 'toto' };
const field = 'dynamic_field';
a?.[field]
```
cf. https://github.com/tc39/proposal-optional-chaining#syntax

Reply:
From your link:

```javascript
a?.[x]  // undefined if `a` is null/undefined, `a[x]` otherwise.
a == null ? undefined : a[x]
```

So I guess:

```javascript
sortedSolution?.[0]?.versions?.[0]?.version
```

should work.

Reply:
For some reason, It didn't work as you describe when I tested before ^^.
Anyway, it works as you describe now and it's indeed much better IMO.